### PR TITLE
Add .arch function

### PR DESCRIPTION
**Pull request by wolfv**
Note: 

- I also had a version that wrote out all options
- We could also implement a stricter type for `Arch`

- I think we should also implement a catch-all platform (`Unknown(String)`) to facilitate adding new platforms easily

### CODE REVIEW

File: crates/rattler_conda_types/src/platform.rs [hunk 1]
```diff
@@ -225,6 +225,23 @@ impl From<Platform> for &'static str {
     }
 }
 
+impl Platform {
+    /// Return the arch string for the platform
+    /// The arch is usually the part after the `-` of the platform string.
+    /// Only for 32 and 64 bit platforms the arch is `x86` and `x86_64` respectively.
+    pub fn arch(&self) -> String {
```
Comment (baszalmstra) [inline crates/rattler_conda_types/src/platform.rs:232]: Can we do this without the unwrap? especially if we get “unknown”. Maybe return option?

Comment (wolfv) [reply crates/rattler_conda_types/src/platform.rs:232]: Good point, especially for `noarch` it should be undefined :)

Comment (wolfv) [reply crates/rattler_conda_types/src/platform.rs:232]: Yeah I do think that's good, but it also makes us have the same data in two places which is what I dislike... but I can write it out and we could add a test or something.

**Comment by wolfv** [comment]
Just for the record, I've also looked at implementing a custom platform (that would be `Custom(String)` or so, but didn't yet, because `String` doesn't implement `Copy` and so we have to refactor a little more code to use `.clone()` in some places.

Or is there a copyable string somewhere that we could/should use?

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**